fusesoc/build/vivado.py

```diff
@@ -0,0 +1,170 @@
+import os.path
+import shutil
+import subprocess
+from fusesoc import utils
+
+from fusesoc.build.backend import Backend
+
+""" Vivado Backend
+
+The Vivado backend executes Xilinx Vivado to build systems and program the FPGA.
+
+The backend defines the following section:
+
+    [vivado]
+    part = <part name> # Format <family><device><package>-<speedgrade>
+    hw_device = <device name> # Format <family><device>_0
+
+A core (usually the system core) can add the following files:
+
+- Standard design sources
+
+- Constraints: Supply xdc files with file_type=xdc
+
+- IP: Supply the IP core xdi file with file_type=xdi and other files (like .prj)
+      as file_type=data
```

Comment:
I would prefer to use one of the existing IP-XACT file types here, like 'use'r or 'unknown'. FYI, file types for latest revision are listed here http://www.accellera.org/images/xmlschema/ipxact/1685-2014/fileType.xsd